internal/client/client.go

```diff
@@ -283,6 +286,42 @@ func InitThriftClient(cfg *config.Config, httpclient *http.Client) (*ThriftServi
 	return tsClient, nil
 }
 
+// handler function for errors returned by the thrift client methods
+func handleClientMethodError(ctx context.Context, err error) dbsqlerr.DBRequestError {
+	if err == nil {
+		return nil
+	}
+
+	// If the passed error indicates an invalid session we inject a bad connection error
+	// into the error stack. This allows the for retrying with a new connection.
+	s := err.Error()
+	if strings.Contains(s, "Invalid SessionHandle") {
```

Comment:
we need to be very careful with session state. If the user expects some state - temp views, etc - and we just open a new session in the background, it may be very hard to understand and debug what is going on.

Reply:
In the context of `database/sql`, if the user is using methods directly on the `sql.DB` object directly (e.g. `db.QueryContext(...)`) then they cannot rely on maintaining per-session state anyway because these methods might create new connections or reuse an arbitrary connection from the pool.

If a user does need to rely on per-session state, then doing so correctly requires explicitly grabbing a connection and then calling methods on the connection. In that case, it's the user's responsibility to manage the connection lifecycle, which includes handling errors and accounting for the possibility that a connection may end up in a bad state.